From 0acce1fb4d7a66732c2db3ca06cb37dded5ec8f1 Mon Sep 17 00:00:00 2001
From: Rajendra Naik <rnayak@ti.com>
Date: Mon, 7 Mar 2011 02:05:15 -0600
Subject: i2c-omap: Program I2C_WE on OMAP4 to enable i2c wakeup

For the I2C module to be wakeup capable, programming I2C_WE
register (which was skipped for OMAP4430) is needed
even on OMAP4.
This fixes i2c controller timeouts which were seen recently
with the static dependency being cleared between MPU and
L4PER clockdomains.

Change-Id: Ifde03dc611e35f0264e958f6385a44c5c1bd8e81
Signed-off-by: Rajendra Nayak <rnayak@ti.com>
Signed-off-by: Rajeev Kulkarni <rajeevk@ti.com>
---
 drivers/i2c/busses/i2c-omap.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-omap.c b/drivers/i2c/busses/i2c-omap.c
index 6e34bc5..d5500b5 100644
--- a/drivers/i2c/busses/i2c-omap.c
+++ b/drivers/i2c/busses/i2c-omap.c
@@ -407,9 +407,7 @@ static int omap_i2c_init(struct omap_i2c_dev *dev)
 			 * REVISIT: Some wkup sources might not be needed.
 			 */
 			dev->westate = OMAP_I2C_WE_ALL;
-			if (dev->rev < OMAP_I2C_REV_ON_4430)
-				omap_i2c_write_reg(dev, OMAP_I2C_WE_REG,
-								dev->westate);
+			omap_i2c_write_reg(dev, OMAP_I2C_WE_REG, dev->westate);
 		}
 	}
 	omap_i2c_write_reg(dev, OMAP_I2C_CON_REG, 0);
-- 
1.6.0.4

